commands/keys.md

```diff
@@ -26,6 +26,8 @@ Supported glob-style patterns:
 
 Use `\` to escape special characters if you want to match them verbatim.
 
+Note that when enabling `Redis cluster-mode`, using a pattern with hash tag which can imply a specific slot can save complexity, as any key matching the pattern will be in the same slot, Redis would only iterate over keys in the slot, rather than the whole database. For example, with the pattern `{a}h*llo`, Redis would only try to match it with the keys in slot 15495, which hash tag `{a}` implies.
```

Comment:
Please add a linebreak between each sentence and avoid very long lines, as described in the styling guideline for this repo (see the README). The single line break has no significance in Markdown anyway.

I think we need to mention that this is an *optimization*. (For KEYS and SCAN it is an optimization. For SORT it is more than an optimization.)

As @soloestoy pointed out, there are some specific rules where the optimization applies which he summarized very well. I just split it up into 3 bullets:

* the pattern must contain a hashtag
* there cannot be wildcards or escape characters before the hashtag
* the hashtag within curly braces cannot contain wildcards or escape characters

The example is very good to clarify these rules:

> For example, `{abc}*` can successfully recognize the hashtag and only scan the slot corresponding to `abc`. However, `*{abc}`, `{a*c}`, or `{a\*bc}` cannot recognize the hashtag.

Then I think @dwdougherty can suggest another way of writing this, perhaps using shorter sentences?

Reply:
I think it's better to add the rules for using hashtags in patterns and the list of supported commands to the [cluster hashtag](https://github.com/redis/redis-doc/blob/master/docs/reference/cluster-spec.md#hash-tags) documentation. This way, all the commands that support using hashtags in patterns can reference this link. Can you help to do this?

Reply:
Done in #2646. Please have a look.